fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java

```diff
@@ -152,46 +99,80 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
-  public void setupNavigationBar() {
-    BottomNavigationView bottomNavigationView = findViewById(R.id.social_media_nav_bar);
-    bottomNavigationView.setOnItemSelectedListener(
+  private void setupNavigationBar() {
+    navbar = findViewById(R.id.social_media_nav_bar);
+    navbar.setOnItemSelectedListener(
         item -> {
           int itemId = item.getItemId();
-          if (itemId == R.id.social_media_home_menu) {
-            mViewModel.openHome();
-          } else if (itemId == R.id.social_media_search_menu) {
-            mViewModel.openSearch();
-          } else if (itemId == R.id.social_media_following_menu) {
-            mViewModel.openFollowing();
-          } else if (itemId == R.id.social_media_profile_menu) {
-            mViewModel.openProfile();
+          if (itemId != mViewModel.getCurrentSelectedItem().getValue()) {
+            mViewModel.setCurrentSelectedItem(itemId);
+          } else {
+            if (itemId == R.id.social_media_home_menu) {
+              Log.d(TAG, "Opening home");
+              openSocialMediaHomeFragment();
+            } else if (itemId == R.id.social_media_search_menu) {
+              Log.d(TAG, "Opening search");
+              openSocialMediaSearchFragment();
+            } else if (itemId == R.id.social_media_following_menu) {
+              Log.d(TAG, "Opening following");
+              openSocialMediaFollowingFragment();
+            } else if (itemId == R.id.social_media_profile_menu) {
+              Log.d(TAG, "Opening profile");
+              openSocialMediaProfileFragment();
+            }
           }
           return true;
         });
   }
 
-  public void setupSocialMediaHomeFragment() {
+  private void subscribeToLaoName() {
+    // Subscribe to "lao name" string
+    mViewModel
+        .getLaoName()
+        .observe(
+            this,
+            newLaoName -> {
+              if (newLaoName != null) {
```

Comment:
In general, since we're doing nothing in the null case, I'd suggest structuring the subscribers as follows:

```
if (X == null) {
  return;
}

// the rest , unindented
```

Reply:
I'll change it. Out of curiosity and to avoid further "mistake", is it a standard somewhere or a personal preference ?

Reply:
It depends on the development environment.

In the automotive industry, for instance, there are some standards that require your functions to have just one exit point (i.e. one `return`), to ensure the flow is easy to follow and without surprises. The static analysis tools in those environments will look at the complexity of the "flow" as a quality criteria.

However, if you think of defensive programming, you'd want to validate all input and abort/exit as fast as possible when something is "off", like here a null value.
There's also another reason I tend to favor (if applicable) this style: it avoids crazy amounts of indentation.
Imagine you have the following (pseudo-)code
```
if x != null {
  if x.IsValid {
    // actual code in this function
  }
  else { 
    log.Error("X is invalid")
  }
}
```

the following is going to be easier to understand / maintain and, more importantly, *write correctly*
```
if x == null {
  return
}

if ! x.IsValid {
  log.Error("X is invalid")
  return
}

// the actual code in this function
```

this becomes especially true when you have more than just 1-2 checks.

Reply:
Thanks for the explanation 👍